src/sentry/relay/config/metric_extraction.py

```diff
@@ -113,14 +111,17 @@ def get_metric_extraction_config(
     sentry_sdk.set_tag("organization_id", project.organization_id)
 
     with sentry_sdk.start_span(op="get_on_demand_metric_specs"):
-        alert_specs, widget_specs = get_on_demand_metric_specs(timeout, project)
+        alert_specs, widget_specs = build_safe_config(
+            "on_demand_metric_specs", get_on_demand_metric_specs, project, default_return=([], [])
+        )  # type: ignore[misc]
     with sentry_sdk.start_span(op="generate_span_attribute_specs"):
-        span_attr_specs = _generate_span_attribute_specs(project)
+        span_attr_specs = build_safe_config(
+            "span_attribute_specs", _generate_span_attribute_specs, project, default_return=[]
+        )
     with sentry_sdk.start_span(op="merge_metric_specs"):
```

Comment:
Assuming the span time is similar, this actually increases the total time a task can take from `1 * Timeout` to `2 * Timeout`, and based on the bumped up timeout we may be exceeding the softtimeout for this task. 

I'd also recommend you change the soft timeout to match the outer bounds of all these checks, just to be safe, confirm with other Relay folks that the increased soft timeout won't cause issues. We'll be bringing these down, but  in the meantime the addition of span specs def needs us to bump up the total task time.

Reply:
Great catch, I have checked with Ingest team, we don't expect generation of span based metrics to take long enough to make problems, so for now we will leave soft time limit as it is, and I will keep an eye on it if it needs to be increased, if some problems happen